astroquery/casda/core.py

```diff
@@ -123,6 +141,248 @@ def _parse_result(self, response, verbose=False):
+        -------
+        The url of the async service and the authenticated id token of the file.
+        """
+        data = BytesIO(response.content)
+
+        votable = parse(data, pedantic=False)
+        results = next(resource for resource in votable.resources if
+                       resource.type == "results")
+        if results is None:
+            return None
+        results_array = results.tables[0].array
+        async_url = None
+        authenticated_id_token = None
+
+        # Find the authenticated id token for accessing the image cube
+        for x in results_array:
+            service_def = x['service_def']
+            if isinstance(service_def, bytes):
+                service_def = service_def.decode("utf8")
+            if service_def == service_name:
+                authenticated_id_token = x['authenticated_id_token']
+                if isinstance(service_def, bytes):
+                    authenticated_id_token = authenticated_id_token.decode("utf8")
+
+        # Find the async url
+        for x in votable.resources:
+            if x.type == "meta":
+                if x.ID == service_name:
+                    for p in x.params:
+                        if p.name == "accessURL":
+                            async_url = p.value
+                            if isinstance(async_url, bytes):
+                                async_url = async_url.decode()
+
+        return async_url, authenticated_id_token
+
+    def _create_soda_job(self, authenticated_id_tokens, soda_url=None):
+        """
+        Creates the async job, returning the url to query the job status and details
+
+        Parameters
+        ----------
+        authenticated_id_tokens: list of str
+            A list of tokens identifying the data products to be accessed.
+        soda_url: str, optional
+            The URL to be used to access the soda service. If not provided, the default CASDA one will be used.
+
+        Returns
+        -------
+        The url of the SODA job.
+        """
+        id_params = list(
+            map((lambda authenticated_id_token: ('ID', authenticated_id_token)),
+                authenticated_id_tokens))
+        async_url = soda_url if soda_url else self._get_soda_url()
+
+        resp = self._request('POST', async_url, params=id_params, cache=False)
+        resp.raise_for_status()
+        return resp.url
+
+    def _run_job(self, job_location, poll_interval=20):
+        """
+        Start an async job (e.g. TAP or SODA) and wait for it to be completed.
+
+        Parameters
+        ----------
+        job_location: str
+            The url to query the job status and details
+        poll_interval: int, optional
+            The number of seconds to wait between checks on the status of the job.
+
+        Returns
+        -------
+        The single word final status of the job. Normally COMPLETED or ERROR
+        """
+        # Start the async job
+        print("Starting the retrieval job...")
```

Comment:
similarly, there should be a non-verbose option here

Reply:
Done.